diff -rupN spacenavd-0.6.orig/src/dev.c spacenavd-0.6/src/dev.c
--- spacenavd-0.6.orig/src/dev.c	2019-12-07 17:26:54.652060096 +0000
+++ spacenavd-0.6/src/dev.c	2019-12-07 17:40:41.537793515 +0000
@@ -199,16 +199,29 @@ static int devid_list[][2] = {
 	{0x46d, 0xc629},	/* space pilot pro*/
 	{0x46d, 0xc62b},	/* space mouse pro*/
 	{0x46d, 0xc640},	/* nulooq */
+	{0x256f, 0xc633},	/* SpaceMouse Enterprise */
 
 	{-1, -1}
 };
 
+static int devid_blacklist[][2] = {
+	{0x256f, 0xc650},	/* CadMouse */
+	{-1, -1}
+ };
+
 static int match_usbdev(const struct usb_device_info *devinfo)
 {
 	int i;
 
-	/* if it's a 3Dconnexion device match it immediately */
+	/* if it's a 3Dconnexion device match it unless blacklisted */
 	if((devinfo->name && strstr(devinfo->name, "3Dconnexion"))) {
+		if(devinfo->vendorid != -1 && devinfo->productid != -1) {
+			for(i=0; devid_blacklist[i][0] > 0; i++) {
+				if(devinfo->vendorid == devid_list[i][0] && devinfo->productid == devid_list[i][1]) {
+					return 0;
+				}
+			}
+		}
 		return 1;
 	}
 
